http.go

```diff
@@ -152,3 +152,13 @@ func (ln tcpKeepAliveListener) Accept() (c net.Conn, err error) {
 	tc.SetKeepAlivePeriod(3 * time.Minute)
 	return tc, nil
 }
+
+func redirectToHTTPS(opts *Options, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I wonder if the `HandlerFunc` needs to check the scheme before it issues a redirect? Could you maybe add a test to check if the scheme is https already that it doesn't issue a redirect?

Reply:
Good callout. I had thought of that originally, but forgot to add it in.